Architecture.md

```diff
@@ -0,0 +1,112 @@
+- an internal application for translating content
+- an application for carriers to manage shippings
+- an application for data and marketing teams to analyze user's behaviour to improve the various services
+- applications that can be embedded in third-party websites
+- a large number of backend services involving full fledged applications or tiny-scoped microservices
+- ...and many others
+
+All of those applications will have to interact and work under the same business domain and one of the core ideas of several domain-centric software designs is to separate the business logic related to the domain (common to all these applications) separately from the actual applications consuming it.
+
+This avoids one common pitfall of many enterprise software where business (or `Core`) logic is first shoved in a specific application and then requires to be refactored away or more commonly cloned across different projects with an exponential increase in software maintenance and velocity of adaptation of software to business needs.
+
+## Entity
+
+An `Entity` encapsulates and abstracts a _business domain object_.
+
+In `effect-ts-app` `Entity`s are defined through `Schema`s defined in `types` packages.
+
+Behaviour, the _what_, _how_, and _when_ we can operate on and with this data is defined in `Core` modules in `api` packages. Behaviour encodes the rules of the business.
+
+An important distinction between `Entity`s and other _business domain objects_ is that `Entity`s **alwyas** have a unique identifier which allows to distinguish each of those from any other `Entity`s of the same type.
+
+`Entity`s can contain and reference other `Entity`s or `Value-Object`s.
+
+An example of an `Entity` can be an `Order` entity in an e-commerce.
+
+By defining `Entity`s encapsulations we can define what `Order`s are (their `Schema`) and how, when and what can be done with them (through their `Core` logic). The business rules encoded in `Core` such as the inability to place `Order`s with a delivery date set in the past, are general and independent from the class of applications that may interface with the `Order`s.
+
+## Value-Object
+
+Similar to `Entity`s, `Value-Object`s also abstract a _business domain object_. The core difference between `Entity`s and `Value-Object`s is that `Value-Object`s do not have a unique identifier.
+`Value-Object`s too are defined by their data shape and their business logic, and those too are defined through `Schema` and `Core` logic inthe same packages where `Entity`s are defined.
+
+An example of a `Value-Object` might be an email or a geographical `Address`. An `Order` might have a delivery `Address` but this `Address` is only defined by its content. `Value-Object`s are generally attached to `Entity`s and modifications of `Value-Object`s are indistinguishable from replacing the `Value-Object`s as they do not hold any unique identifier and their identity is defined by their content only.
```

Comment:
an Email is generally a refined primitive string, not a value **Object**.

Reply:
So what would be an example of a `Value-Object`?

Reply:
@enricopolanski usually Address, defined as e.g addressfield1, 2, 3, street, number, zip, etc.